File: bug29_variable_reset_inside_loop/solved_main.js
function run() {
    let acc=0n;   // FIX: acc=0n belongs here before the loop
    for(let i=0n;i<10n;i++){
        // FIX: removed "acc = 0n" that was incorrectly placed inside the loop
        acc^=i*0x6C62272E07BB0142n;
        acc=BigInt.asUintN(64,acc*0x9E3779B97F4A7C15n);
        let sl=i%13n+1n, sr=63n-i%13n;
        acc=BigInt.asUintN(64,(acc<<sl)|(acc>>sr));
        acc^=acc>>31n;
    }
    let buf=Buffer.alloc(8);
    buf.writeBigUInt64BE(acc,0);
    console.log(buf.toString('base64'));
}
run();

File: bug29_variable_reset_inside_loop/solved_main.py
import base64, struct

def run():
    acc = 0   # FIX: acc = 0 belongs HERE (before loop), not inside the loop
    for i in range(10):
        # FIX: removed "acc = 0" that was incorrectly placed here inside the loop
        acc ^= i * 0x6C62272E07BB0142
        acc = (acc * 0x9E3779B97F4A7C15) & 0xFFFFFFFFFFFFFFFF
        acc = ((acc << (i % 13 + 1)) | (acc >> (63 - i % 13))) & 0xFFFFFFFFFFFFFFFF
        acc ^= acc >> 31
    raw = struct.pack(">Q", acc)
    print(base64.b64encode(raw).decode())

run()
